operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/DynamicallyCreatedDependentResource.java

```diff
@@ -41,19 +42,19 @@
   Map<String, R> getSecondaryResources(P primary, Context<P> context);
 
   /**
-   * Used to delete resource if the desired count is lower than the actual count of a resource.
+   * Deletes the actual resource if it's not in the set of desired secondary resources for the
+   * specified primary.
    *
-   * @param primary resource
-   * @param resource actual resource from the cache for the index
-   * @param key key of the resource
+   * @param primary the primary resource for which we want to remove now undesired secondary
+   *        resources still present on the cluster
+   * @param resource the actual resource existing on the cluster that needs to be removed
    * @param context actual context
    */
-  void deleteBulkResource(P primary, R resource, String key, Context<P> context);
+  void deleteTargetResource(P primary, R resource, Context<P> context);
```

Comment:
But not sure we want to remove the key. I'm more to more complete APIs, so that is an information that some might miss if providing a custom implementation. So this way it's more future proof, also integral part.